src/cli/cmd/apply/mod.rs

```diff
@@ -0,0 +1,200 @@
+mod home_manager;
+#[cfg(target_os = "macos")]
+mod nix_darwin;
+#[cfg(target_os = "linux")]
+mod nixos;
+
+use std::{
+    os::unix::prelude::PermissionsExt,
+    path::PathBuf,
+    process::{ExitCode, Stdio},
+};
+
+use clap::{Parser, Subcommand};
+use color_eyre::eyre::Context;
+
+use crate::{
+    cli::{
+        cmd::{nix_command, parse_flake_output_ref},
+        error::FhError,
+    },
+    path,
+};
+
+use self::home_manager::{HomeManager, HOME_MANAGER_SCRIPT};
+
+#[cfg(target_os = "linux")]
+use self::nixos::{NixOs, NIXOS_PROFILE, NIXOS_SCRIPT};
+
+#[cfg(target_os = "macos")]
+use self::nix_darwin::{NixDarwin, DARWIN_REBUILD_ACTION, NIX_DARWIN_PROFILE, NIX_DARWIN_SCRIPT};
+
+use super::{CommandExecute, FlakeHubClient};
+
+/// Apply the configuration at the specified FlakeHub output reference to the current system
+#[derive(Parser)]
+pub(crate) struct ApplySubcommand {
+    #[clap(subcommand)]
+    system: System,
+
+    #[clap(from_global)]
+    api_addr: url::Url,
+}
+
+#[derive(Subcommand)]
+enum System {
+    /// Resolve the store path for a Home Manager configuration and run its activation script
+    HomeManager(HomeManager),
+
+    #[cfg(target_os = "macos")]
+    /// Resolve the store path for a nix-darwin configuration and run its activation script
+    NixDarwin(NixDarwin),
+
+    #[cfg(target_os = "linux")]
+    /// Apply the resolved store path on a NixOS system
+    #[clap(name = "nixos")]
+    NixOs(NixOs),
+}
+
+#[async_trait::async_trait]
+impl CommandExecute for ApplySubcommand {
+    async fn execute(self) -> color_eyre::Result<ExitCode> {
+        let output_ref = match &self.system {
+            System::HomeManager(home_manager) => home_manager.output_ref()?,
+            #[cfg(target_os = "linux")]
+            System::NixOs(nixos) => nixos.output_ref()?,
+            #[cfg(target_os = "macos")]
+            System::NixDarwin(nix_darwin) => nix_darwin.output_ref()?,
+        };
+
+        tracing::info!("Resolving store path for output: {}", output_ref);
+
+        let output_ref = parse_flake_output_ref(&output_ref)?;
+
+        let resolved_path = FlakeHubClient::resolve(self.api_addr.as_ref(), &output_ref).await?;
+
+        tracing::debug!(
+            "Successfully resolved reference {} to path {}",
+            &output_ref,
+            &resolved_path.store_path
+        );
+
+        match self.system {
+            System::HomeManager(_) => {
+                // /nix/store/{path}/activate
+                let script_path = path!(&resolved_path.store_path, HOME_MANAGER_SCRIPT);
```

Comment:
I think elsewhere we're downloading the closure in `apply_path_to_profile`, which we're not running here. We don't run it here, because HomeManager takes care of it. But I think that means we haven't actually downloaded it in the HM case.